link

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env php
+<?php
+
+/*
+* This file is part of the Symfony package.
+*
+* (c) Fabien Potencier <fabien@symfony.com>
+*
+* For the full copyright and license information, please view the LICENSE
+* file that was distributed with this source code.
+*/
+
+/**
+ * Links dependencies to components to a local clone of the main symfony/symfony GitHub repository.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+
+if (2 !== $argc) {
+    echo 'Link dependencies to components to a local clone of the main symfony/symfony GitHub repository.' . PHP_EOL . PHP_EOL;
+    echo 'Usage: ./link /path/to/the/project' . PHP_EOL;
+    exit(1);
+}
+
+if (!is_dir("$argv[1]/vendor/symfony")) {
+    echo "The directory \"$argv[1]\" does not exist or the dependencies are not installed, did you forget to run \"composer install\" in your project?".PHP_EOL;
+    exit(1);
+}
+
+$sfPackages = array();
+foreach (glob(__DIR__.'/src/Symfony/{Bundle,Bridge,Component}/*', GLOB_BRACE | GLOB_ONLYDIR | GLOB_NOSORT) as $dir) {
+    $sfPackages[json_decode(file_get_contents("$dir/composer.json"))->name] = $dir;
+}
+
+foreach (glob("$argv[1]/vendor/symfony/*", GLOB_ONLYDIR | GLOB_NOSORT) as $dir) {
+    $package = 'symfony/'.basename($dir);
+    if (is_link($dir)) {
+        echo "\"$package\" is already a symlink, skipping.".PHP_EOL;
+        continue;
+    }
+
+    if (!isset($sfPackages[$package])) {
```

Comment:
maybe add a message here: "Skipping ..., not a symfony/symfony package."?
also, should we make the script work with symfony/symfony? I'm sure it would help standard edition users.
Also wondering: where should we tell ppl that this script exists and when it should be used?

Reply:
:+1: for symlinking symfony/symfony as well. I think it fits the [contributing](https://symfony.com/doc/current/contributing/index.html) section. That's what i'd use it for... cloning  my symfony fork (so symlinks pointing here), adding symfony as 2nd upstream remote.

Reply:
👎 for this "Skipping .. not a symfony package", it would be far too verbose and the output would be unreadable IMHO. If you really want to know, as a user, put a `echo` in the PHP script or something I'd say.

👍 for symfony/symfony tho

Reply:
You can already rely on `composer install -prefer-source` for `symfony/symfony`.

Reply:
given this tool requires you to have a centralized clone somewhere, it's tempting to link it to multiple projects and switch branches on the fly :) i dont know if thats a supported use case, but if so symfony/symfony should be detected IMHO.

Reply:
> You can already rely on composer install --prefer-source for symfony/symfony.
  
I do as @ro0NL: one repo for dev, never touch directly inside vendor